lib/PesyConf.re

```diff
@@ -658,18 +660,75 @@ let toPesyConf = (projectPath, rootName, pkg) => {
       | JSON.NullJSONValue () => None
       | e => raise(e)
       };
+
+    let stubsVersionGuard = (~cnJSON, ~fsJSON, prop, version) => {
+      switch (
+        prop,
+        List.hd(String.split_on_char('.', version)),
+        JSON.isNull(cnJSON),
+        JSON.isNull(fsJSON),
+      ) {
+      | ("cNames", "2", false, true) => raise(VersionMismatch_CNames)
+      | ("foreignStubs", "1", true, false) =>
+        raise(VersionMismatch_ForeignStubs)
+      | ("cNames", "1", false, true) => cnJSON
+      | ("cNames", "1", false, false) => cnJSON
+      | ("foreignStubs", "2", true, false) => fsJSON
+      | ("foreignStubs", "2", false, false) => fsJSON
+      | (_, _, _, _) => JSON.jsonNullValue()
+      };
+    };
```

Comment:
@ManasJayanth this doesn't feel right, is there a better way to do this?

Reply:
One way to deal with them is to turn strings into a type.

```re
type a = ForeignStubs | CStubs
let parse: string /* unvalidated raw string from a file */ => a
```

Alexis King has a great blog post on this
https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/